gpt_index/indices/query/query_combiner/base.py

```diff
@@ -27,13 +24,17 @@ def __init__(
         self,
         index_struct: IndexStruct,
         query_transform: BaseQueryTransform,
+        query_runner: Any, # NOTE: type as Any to avoid circular dependency
     ) -> None:
         """Init params."""
         self._index_struct = index_struct
         self._query_transform = query_transform
+        from gpt_index.indices.query.query_runner import QueryRunner
+        assert isinstance(query_runner, QueryRunner)
+        self._query_runner = query_runner
 
     @abstractmethod
-    def run(self, query_obj: BaseGPTIndexQuery, query_bundle: QueryBundle) -> Response:
+    def run(self, query_bundle: QueryBundle, level: int = 0) -> Response:
```

Comment:
Note: i added `arun` in https://github.com/jerryjliu/llama_index/pull/845 in order to support async use cases